src_kt/java/arcs/storage/driver/RamDisk.kt

```diff
@@ -12,20 +12,42 @@
 package arcs.storage.driver
 
 import arcs.storage.Driver
-import arcs.storage.ExistenceCriteria
 import arcs.storage.DriverFactory
 import arcs.storage.DriverProvider
+import arcs.storage.ExistenceCriteria
 import arcs.storage.StorageKey
+import arcs.storage.StorageKeyParser
 
 /** Protocol to be used with the ramdisk driver. */
 const val RAMDISK_DRIVER_PROTOCOL = "ramdisk"
 
 /** Storage key for a piece of data managed by the ramdisk driver. */
 data class RamDiskStorageKey(private val unique: String) : StorageKey(RAMDISK_DRIVER_PROTOCOL) {
-  override fun toKeyString(): String = unique
+    override fun toKeyString(): String = unique
+
+    override fun childKeyWithComponent(component: String): StorageKey =
+        RamDiskStorageKey("$unique/$component")
+
+    override fun toString(): String = super.toString()
```

Comment:
Is this needed? Isn't that the default behaviour?

Reply:
Unfortunately it is needed. I think it has something to do with data classes implementing `toString` for you, which - while normally helpful - isn't really what we want here.